### ISSUE TITLE
Add allow / block list for global helper functions which can be overridden by extensions

### ISSUE THREAD

**Issue by saxena-anurag** [microsoft/ebpf-for-windows#1791]
Add support for a `allow` or `block` list of global helper functions which can be overloaded by extension.

For example, we would not want extensions to overload map related helper functions.

**Comment by dthaler** [comment]
I'm not yet convinced we would not want extensions to overload map related helper functions... what if an extension wanted to add a new map type?   It could do that with an override.   Otherwise we'd need a way to let extensions add map types, which we don't have yet.

**Comment by saxena-anurag** [comment]
> I'm not yet convinced we would not want extensions to overload map related helper functions... what if an extension wanted to add a new map type? It could do that with an override. Otherwise we'd need a way to let extensions add map types, which we don't have yet.

Map operations was just an example that I mentioned. Per our earlier discussion, there was a concern on what if ebpfcore _does not_ want the extensions to override some specific global helper functions. Adding the allow / block list will let ebpfcore control which global helper functions can be overridden.

**Comment by dthaler** [comment]
I agree it would.  I just don't know of a reason to do it now, so we should hold off doing anything until we have a need.  If we never have a need, then this issue could just be closed.

**Comment by saxena-anurag** [comment]
> I agree it would. I just don't know of a reason to do it now, so we should hold off doing anything until we have a need. If we never have a need, then this issue could just be closed. Maybe this should be converted to a discussion and only converted back to an issue if/when there is a need?

Agree that there is no use case right now. We can also probably move this issue to backlog and revisit if/when needed?
